Makefile docker-base target missing newline interpretation

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Hello there! 😄

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

Please do not remove sections of this template when submitting your issue.  If the section is not relevant to your issue, leave the section but enter "n/a" for not applicable.

Note some sections are marked "REQUIRED" and must be provided.  Never mark required sections "n/a".

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->


# 🐞 Bug Report

### Affected Services [**REQUIRED**]
<!-- Can you pin-point one or more EdgeX services (Data, Metadata, Command, etc...) as the source of the bug? -->
<!-- ✍️edit: --> The issue is located in Makefile target `docker_base` when building custom images like `make docker-nats` locally as instructed by README.

### Is this a regression?

<!-- Did this behavior work correctly in the previous version? -->
<!-- ✍️--> No, the issue was there when the commit that introduced the code was integrated:
```
commit eebbc51e4a300ba4ec20fbfde4689b2c90fa3bc0
Author: Ernesto Ojeda <ernesto.ojeda@intel.com>
Date:   Wed May 25 11:41:11 2022 -0700
```

### Description and Minimal Reproduction [**REQUIRED**]

The `docker_base` make target is using plain `echo` to generate an inline Dockerfile and pipe it to `docker`, however newlines are not being correctly interpreted there because `echo` without any options does not interpret them, but print then instead. Executing any target that depends on `docker_base`, like `docker-nats`, hits the issue.

## 🔥 Exception or Error
<pre><code>
make -e ADD_BUILD_TAGS=include_nats_messaging docker
make[1]: Entering directory '/home/bmeneg/git/volis/edgex-go'
echo "Building local cache image";\
response=200; \
if [ "${response}" = "200" ]; then \
        echo "Found base Dockerfile"; \
        curl -s "https://raw.githubusercontent.com/edgexfoundry/ci-build-images/golang-1.23/Dockerfile" | docker build -t edgex-go-local-cache-base -f - .; \
        echo "FROM edgex-go-local-cache-base\nWORKDIR /edgex-go\nCOPY go.mod .\nRUN go mod download" | docker build -t edgex-go-local-cache -f - .; \
else \
        echo "No base Dockerfile found. Using golang:1.23-alpine"; \
        echo "FROM golang:1.23-alpine\nRUN apk add --update make git\nWORKDIR /edgex-go\nCOPY go.mod .\nRUN go mod download" | docker build -t edgex-go-local-cache -f - .; \
fi
Building local cache image
Found base Dockerfile
[+] Building 2.0s (10/10) FINISHED                                                                      docker:default
 => [internal] load build definition from Dockerfile                                                              0.0s
 => => transferring dockerfile: 1.30kB                                                                            0.0s
 => [internal] load metadata for docker.io/library/golang:1.23-alpine3.22                                         1.7s
 => [internal] load .dockerignore                                                                                 0.0s
 => => transferring context: 109B                                                                                 0.0s
 => [internal] load build context                                                                                 0.2s
 => => transferring context: 94B                                                                                  0.2s
 => [1/5] FROM docker.io/library/golang:1.23-alpine3.22@sha256:383395b794dffa5b53012a212365d40c8e37109a626ca30d6  0.0s
 => CACHED [2/5] COPY ./.golangci.yml /etc/.golangci.yml                                                          0.0s
 => CACHED [3/5] RUN if [ $(uname -m) == "x86_64" ]; then wget -O /usr/local/bin/hadolint https://github.com/had  0.0s
 => CACHED [4/5] RUN apk add --update --no-cache make git curl bash pkgconfig build-base linux-headers musl-dev   0.0s
 => CACHED [5/5] RUN go install github.com/gotesttools/gotestfmt/v2/cmd/gotestfmt@latest                          0.0s
 => exporting to image                                                                                            0.0s
 => => exporting layers                                                                                           0.0s
 => => writing image sha256:9ce179ebf62314c86956077c3c4327f448ff04fe4775f500cabcb7b0d9007dd2                      0.0s
 => => naming to docker.io/library/edgex-go-local-cache-base                                                      0.0s
[+] Building 0.1s (1/1) FINISHED                                                                        docker:default
 => [internal] load build definition from Dockerfile                                                              0.0s
 => => transferring dockerfile: 190B                                                                              0.0s
 => WARN: FromAsCasing: '/edgex-go\nCOPY' and 'FROM' keywords' casing do not match (line 1)                       0.0s

 1 warning found (use docker --debug to expand):
 - FromAsCasing: '/edgex-go\nCOPY' and 'FROM' keywords' casing do not match (line 1)
Dockerfile:1
--------------------
   1 | >>> FROM edgex-go-local-cache-base\nWORKDIR /edgex-go\nCOPY go.mod .\nRUN go mod download
   2 |
--------------------
ERROR: failed to build: failed to solve: dockerfile parse error on line 1: FROM requires either one or three arguments
make[1]: *** [Makefile:201: docker_base] Error 1
make[1]: Leaving directory '/home/bmeneg/git/volis/edgex-go'
make: *** [Makefile:192: docker-nats] Error 2
</code></pre>


## 🌍  Your Environment
**Deployment Environment:**
Fedora 42 Workstation
GNU bash, version 5.2.37
GNU Make 4.4.1

**EdgeX Version [**REQUIRED**]:** Any version since 2022, when the code was inserted

**Anything else relevant?**
I'm planning to send a PR to fix it.

--- a/Makefile
+++ b/Makefile
@@ -203,10 +203,10 @@ docker_base:
 	if [ "$${response}" = "200" ]; then \
 		echo "Found base Dockerfile"; \
 		curl -s "$(BASE_DOCKERFILE)" | docker build -t $(LOCAL_CACHE_IMAGE_BASE) -f - .; \
-		echo "FROM $(LOCAL_CACHE_IMAGE_BASE)\nWORKDIR /edgex-go\nCOPY go.mod .\nRUN go mod download" | docker build -t $(LOCAL_CACHE_IMAGE) -f - .; \
+		printf "FROM $(LOCAL_CACHE_IMAGE_BASE)\nWORKDIR /edgex-go\nCOPY go.mod .\nRUN go mod download" | docker build -t $(LOCAL_CACHE_IMAGE) -f - .; \
 	else \
 		echo "No base Dockerfile found. Using golang:$(GO_VERSION)-alpine"; \
-		echo "FROM golang:$(GO_VERSION)-alpine\nRUN apk add --update make git\nWORKDIR /edgex-go\nCOPY go.mod .\nRUN go mod download" | docker build -t $(LOCAL_CACHE_IMAGE) -f - .; \
+		printf "FROM golang:$(GO_VERSION)-alpine\nRUN apk add --update make git\nWORKDIR /edgex-go\nCOPY go.mod .\nRUN go mod download" | docker build -t $(LOCAL_CACHE_IMAGE) -f - .; \
 	fi
 
 dcore: dmetadata ddata dcommand

